MinimedKit/PumpManager/MinimedPumpManager.swift

```diff
@@ -659,29 +669,48 @@ public class MinimedPumpManager: RileyLinkPumpManager, PumpManager {
                         }
                         return
                     } catch let error {
+                        self.bolusState = .none
                         completion(.failure(error))
                         return
                     }
                 }
                 
                 let date = Date()
-                var deliveryTime = TimeInterval(minutes: units / MinimedPumpManager.deliveryUnitsPerMinute)
-                if self.state.pumpModel.constrainsBolusDeliveryTimeTo5Minutes {
-                    deliveryTime = min(TimeInterval(minutes: 5), deliveryTime)
-                }
+                let deliveryTime = self.state.pumpModel.bolusDeliveryTime(units: units)
                 let endDate = date.addingTimeInterval(deliveryTime)
                 let dose = DoseEntry(type: .bolus, startDate: date, endDate: endDate, value: units, unit: .units)
                 willRequest(dose)
 
                 try session.setNormalBolus(units: units)
-                completion(.success(dose))
+
+                // Between bluetooth and the radio and firmware, about 2s on average passes before we start tracking
+                let commsOffset = TimeInterval(seconds: -2)
```

Comment:
So, you mean the bolus actually started two seconds ago (on average)?

Reply:
Yes